### DISCUSSION TITLE
can cmd ask for link

### DISCUSSION THREAD

**Discussion by Trashycount** [streamlink/streamlink#4917] [Q&A] [answered]
Would there be a way to make it where I could open this: streamlink -o "C:\{title}{author}.mkv" [STREAM] best and make it where it asks for the stream link and i could copy it into the cmd prompt instead of having it already in there?

**Comment by bastimeyer** [comment, marked as answer]
I don't know if I fully understand what you're trying to say, but it sounds like you're looking for a wrapper shell script which prompts for the link, right?

**launch.sh** (BASH)
```sh
#!/usr/bin/env bash
read -p "URL: " url
streamlink "${@}" --output "/some/path/{author} - {title}.ts" "${url}" best
```

**launch.bat** (Windows batch script - CMD)
```batch
@echo off
set /p "url=URL: "
streamlink %* --output "C:\some\path\{author} - {title}.ts" "%url%" best
```

**launch.ps1** (Powershell)
```pwsh
#!/usr/bin/env pwsh
$url = Read-Host "URL"
streamlink $args --output "C:\some\path\{author} - {title}.ts" $url best
```
